snuba/datasets/events.py

```diff
@@ -355,23 +357,26 @@ def _tags_expr(self, column_name, body):
         col, k_or_v = column_name.split('_', 1)
         nested_tags_only = state.get_config('nested_tags_only', 1)
 
+        qualified_col = qualified_column(col, table_alias)
         # Generate parallel lists of keys and values to arrayJoin on
         if nested_tags_only:
-            key_list = '{}.key'.format(col)
-            val_list = '{}.value'.format(col)
+            key_list = '{}.key'.format(qualified_col)
+            val_list = '{}.value'.format(qualified_col)
         else:
             promoted = self._get_promoted_columns()[col]
             col_map = self._get_column_tag_map()[col]
             key_list = u'arrayConcat([{}], {}.key)'.format(
                 u', '.join(u'\'{}\''.format(col_map.get(p, p)) for p in promoted),
-                col
+                qualified_col
             )
             val_list = u'arrayConcat([{}], {}.value)'.format(
                 ', '.join(string_col(self, p) for p in promoted),
-                col
+                qualified_col
             )
 
-        cols_used = all_referenced_columns(body) & set(['tags_key', 'tags_value'])
+        qualified_key = qualified_column("tags_key", table_alias)
+        qualified_value = qualified_column("tags_value", table_alias)
+        cols_used = all_referenced_columns(body) & set([qualified_key, qualified_value])
```

Comment:
Is the assumption here that all of the columns in the body with `all_referenced_columns` are already qualified?

Reply:
Yes. I would start with a strict syntax: in order to use join, everything has to be qualified. Then we can relax it if it makes sense. I will try to add an error condition to enforce this condition.

Reply:
Makes sense, thanks for the clarification. Probably worth asserting or at least commenting that here if you haven't already.